### PR TITLE
Add help messages for ms mp mL mo commands

### PR DESCRIPTION
**Pull request by alex-bender**
https://github.com/radareorg/radare2/issues/19968
The `mk?` left unfixed. Looks like it is absent now. 

- [x] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

File: libr/core/cmd_mount.c [hunk 1]
```diff
@@ -32,6 +32,31 @@ static const char *help_msg_mf[] = {
 	NULL
 };
 
+static const char *help_msg_md[] = {
+	"Usage:", "md /", "list directory contents for path",
```
Comment (trufae) [inline libr/core/cmd_mount.c:36]:
```suggestion
	"Usage:", "md /", "list files and directory on the virtual r2's fs",
	"md", " /", "list directory on the root directory",
	"md", "", "list files and directories in current directory",
```

File: libr/core/cmd_mount.c [hunk 2]
```diff
@@ -239,16 +260,28 @@ static int cmd_mount(void *data, const char *_input) {
 		}
 		break;
 	case 'L': // "mL" list of plugins
+		if (input[1] == '?') { // "mL?"
+			r_core_cmd_help (core, help_msg_mL);
+			break;
+		}
```
Comment (trufae) [inline libr/core/cmd_mount.c:266]: Else instead of break

**Review by trufae — CHANGES_REQUESTED** [review]
@Lazula did an api to show the help grepping with a word. Maybe we can use that here if the subcommand helps dont provide much extra subcommands

**Comment by Lazula** [comment]
I recommend that you edit `help_msg_m` instead of creating new messages. Then you can use `r_core_cmd_help_match_spec(core, help_msg_m, "m", <*input, input[1], etc.>, true);`, but you will need to do this before advancing the input pointer in each command.

If you want help for one specific command, this is a better way to do it. Separate help messages ought to only be used when there are subcommands to list. Doing it this way also prevents clutter in recursive help.

File: libr/core/cmd_mount.c [hunk 3]
```diff
@@ -239,16 +235,26 @@ static int cmd_mount(void *data, const char *_input) {
 		}
 		break;
 	case 'L': // "mL" list of plugins
+		if (input[1] == '?') { // "mL?"
+			r_core_cmd_help_match_spec(core, help_msg_m, "mL", 0, true);
```
Comment (trufae) [inline libr/core/cmd_mount.c:239]:
```suggestion
			r_core_cmd_help_match_spec (core, help_msg_m, "mL", 0, true);
```

File: libr/core/cmd_mount.c [hunk 4]
```diff
@@ -239,16 +235,26 @@ static int cmd_mount(void *data, const char *_input) {
...
+		}
 		r_list_foreach (core->fs->plugins, iter, plug) {
 			r_cons_printf ("%10s  %s\n", plug->name, plug->desc);
 		}
 		break;
 	case 'l': // "ml"
 	case 'd': // "md"
+		if (input[1] == '?') { // "md?"
+			r_core_cmd_help_match_spec(core, help_msg_m, "md", 0, true);
```
Comment (trufae) [inline libr/core/cmd_mount.c:248]:
```suggestion
			r_core_cmd_help_match_spec (core, help_msg_m, "md", 0, true);
```

File: libr/core/cmd_mount.c [hunk 5]
```diff
@@ -239,16 +235,26 @@ static int cmd_mount(void *data, const char *_input) {
...
+		}
 		cmd_mount_ls (core, input + 1);
 		break;
-	case 'p':
+	case 'p': // "mp"
 		input = (char *)r_str_trim_head_ro (input + 1);
+		if (input[0] == '?') { // "mp?"
+			r_core_cmd_help_match_spec(core, help_msg_m, "mp", 0, true);
```
Comment (trufae) [inline libr/core/cmd_mount.c:255]:
```suggestion
			r_core_cmd_help_match_spec (core, help_msg_m, "mp", 0, true);
```

File: libr/core/cmd_mount.c [hunk 6]
```diff
@@ -268,6 +274,9 @@ static int cmd_mount(void *data, const char *_input) {
 		break;
 	case 'o': // "mo"
 		input = (char *)r_str_trim_head_ro (input + 1);
+		if (*input == '?') { // "mo?"
+			r_core_cmd_help_match_spec(core, help_msg_m, "mo", 0, true);
```
Comment (trufae) [inline libr/core/cmd_mount.c:278]:
```suggestion
			r_core_cmd_help_match_spec (core, help_msg_m, "mo", 0, true);
```

File: libr/core/cmd_mount.c [hunk 7]
```diff
@@ -426,6 +435,10 @@ static int cmd_mount(void *data, const char *_input) {
 			return false;
 		}
 		input = (char *)r_str_trim_head_ro (input + 1);
+		if (*input == '?') { // "ms?"
+			r_core_cmd_help_match_spec(core, help_msg_m, "ms", 0, true);
```
Comment (trufae) [inline libr/core/cmd_mount.c:439]:
```suggestion
			r_core_cmd_help_match_spec (core, help_msg_m, "ms", 0, true);
```

**Review by trufae — APPROVED** [review]

**Pull request merged**